python-package/xgboost/core.py

```diff
@@ -1553,17 +1602,33 @@ def predict(self,
             untransformed margin value of the prediction. Note the last row and
             column correspond to the bias term.
 
-        validate_features : bool
+        validate_features :
             When this is True, validate that the Booster's and data's
             feature_names are identical.  Otherwise, it is assumed that the
             feature_names are the same.
 
-        training : bool
+        training :
             Whether the prediction value is used for training.  This can effect
             `dart` booster, which performs dropouts during training iterations.
 
             .. versionadded:: 1.0.0
 
+        iteration_range :
+            Specifies which layer of trees are used in prediction.  For example, if a
+            random forest is trained with 100 rounds.  Specifying `iteration_range=(10,
+            20)`, then only the forests built during [10, 20) (half open set) rounds are
+            used in this prediction.
+
+            .. versionadded:: 1.4.0
+
+        strict_shape :
```

Comment:
Can you provide a bit more context on why this parameter is needed? Keeping in mind that users can call ndarray.reshape() on the output if they require a strict shape.

Reply:
Yup.  It's used to clarify the dimensions and make it consistent.  Just to give an example, so on top of mind, could you name the dimension of predict leaf?  Also how many dimensions of softmax should have with and without output margin?

Reply:
At some point we might lower the calculation down into `Learner` class instead of C API.  Right now I just want to remove the burden from users.